rclcpp/src/rclcpp/context.cpp

```diff
@@ -34,8 +35,27 @@ static std::vector<std::weak_ptr<rclcpp::Context>> g_contexts;
 
 using rclcpp::Context;
 
+static
+std::shared_ptr<std::mutex>
+get_global_logging_configure_mutex()
+{
+  static auto mutex = std::make_shared<std::mutex>();
+  return mutex;
+}
+
+static
+size_t &
+get_logging_reference_count()
+{
+  static size_t ref_count = 0;
+  return ref_count;
+}
```

Comment:
@ivanpauno meta: I won't block this PR on it, but I _think_ this reference counting should be down in `rcl`. We haven't explored multi-language ROS2 executables much yet, but this could be a problem if we (or anyone) wanted to go down that path.

Reply:
IMHO, no.
We probably want to drop the message count in the future and have a more elegant solution, and so I would avoid adding a reference count API in `rcl`.

Reply:
I see. Then if this aims to be just a temporary solution, a TODO and/or an follow-up issue would be nice.